src/app/application/components/data-source/DataLoader.vue

```diff
@@ -53,55 +53,60 @@
     <template v-else>
       <slot
         name="loadable"
-        :data="props.src !== '' ? allData[0] : undefined"
-        :error="props.src !== '' ? allErrors[0] : undefined"
+        :data="srcData"
+        :error="srcError"
         :refresh="props.src !== '' ? refresh : () => {}"
       />
       <slot
         v-if="props.loader && typeof slots.loadable === 'undefined'"
         name="connecting"
-        :data="props.src !== '' ? allData[0] : undefined"
-        :error="props.src !== '' ? allErrors[0] : undefined"
+        :data="srcData"
+        :error="srcError"
         :refresh="props.src !== '' ? refresh : () => {}"
       >
         <LoadingBlock />
       </slot>
       <slot
         v-else
         name="default"
-        :data="props.src !== '' ? allData[0] : undefined"
-        :error="props.src !== '' ? allErrors[0] : undefined"
+        :data="srcData as NonNullable<typeof srcData>"
+        :error="srcError"
```

Comment:
I figured that you never use the exported data from `DataLoader` unless you have provided a `:src` (if you provide `data="[...]"` you already have access to the data in order to provide it in the first place). The DataLoader `error` state needs to use `allErrors[0]` though so you get a reference to the first error in a potential list of `:errors=[...]` that failed.